Sources_v3/StreamChatUI/MessageActionsPopup/MessageActionsView.swift

```diff
@@ -0,0 +1,69 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class MessageActionsView<ExtraData: UIExtraDataTypes>: View, UIConfigProvider {
+    public var actionItems: [ChatMessageActionItem] = [] {
+        didSet { updateContent() }
+    }
+
+    // MARK: Subviews
+
+    public private(set) lazy var stackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.axis = .vertical
+        stackView.alignment = .fill
+        stackView.distribution = .fill
+        stackView.spacing = 1
+        return stackView.withoutAutoresizingMaskConstraints
+    }()
+
+    // MARK: Overrides
+
+    override open func setUpAppearance() {
+        layer.cornerRadius = 16
+        layer.masksToBounds = true
+    }
+
+    override open func setUpLayout() {
+        embed(stackView)
+    }
+
+    override open func defaultAppearance() {
+        backgroundColor = .outgoingMessageBubbleBorder
+    }
+
+    override open func updateContent() {
+        stackView.arrangedSubviews.forEach {
+            $0.removeFromSuperview()
+            stackView.removeArrangedSubview($0)
+        }
+
+        actionItems.forEach {
+            let actionView = uiConfig.messageList.messageActionButton.init()
+            actionView.actionItem = $0
+            stackView.addArrangedSubview(actionView)
+        }
+    }
+}
+
+// MARK: - Controller
+
+open class ChatMessageActionsViewController<ExtraData: UIExtraDataTypes> {
+    // MARK: - Properties
+
+    public let view: MessageActionsView<ExtraData>
+
+    // MARK: - Init
+
+    public init(
+        view: MessageActionsView<ExtraData>,
+        availableActions: [ChatMessageActionItem]
+    ) {
+        self.view = view
+        self.view.actionItems = availableActions
+    }
+}
```

Comment:
I get the idea of having `ViewController` in the sense of a "controller" of a "view" but I'd strongly recommend not using it for types that are not subclasses of `UIViewController`. It will be extremely confusing for our customers.

Reply:
Any reason why you don't want to make it a `UIViewController` subclass?

Reply:
As discussed in Slack, controllers are updated to inherit from `UIViewController` in 7097ed8282ad0712fb5372836dba3c370639c30d and 498372f6ca35c4905742a6142443d28fba092b12